components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java

```diff
@@ -98,13 +104,15 @@ public Secret getSecret(String secretName) throws SecretManagementException {
     }
 
     @Override
-    public Secrets getSecrets() throws SecretManagementException {
+    public Secrets getSecrets(String secretTypeName) throws SecretManagementException {
 
         validateSecretManagerEnabled();
-        List secretList = this.getSecretDAO().getSecrets(getTenantId());
+        validateSecretsRetrieveRequest(secretTypeName);
+        SecretType secretType = getSecretType(secretTypeName);
+        List secretList = this.getSecretDAO().getSecrets(secretType, getTenantId());
         if (secretList == null) {
             if (log.isDebugEnabled()) {
-                log.debug("No secret found for the tenant: " + getTenantDomain());
+                log.debug("No secret found the secretTypeName: " + secretTypeName + "for the tenant: " + getTenantDomain());
```

Comment:
`log.debug("No secret found for the secretTypeName: " + secretTypeName + "for the tenant: " + getTenantDomain());`

Reply:
Addressed with commit [001ee25](https://github.com/wso2/carbon-identity-framework/pull/3762/commits/001ee2501f3eab862efd07a13864e81939cfe648)